pkg/groupsnapshot/controllers/groupsnapshot.go

```diff
@@ -205,24 +208,25 @@ func (m *GroupSnapshotController) handleInitial(groupSnap *stork_api.GroupVolume
 	return updateCRD, err
 }
 
-func (m *GroupSnapshotController) handlePreSnap(groupSnap *stork_api.GroupVolumeSnapshot) (bool, error) {
+func (m *GroupSnapshotController) handlePreSnap(groupSnap *stork_api.GroupVolumeSnapshot) (
+	*stork_api.GroupVolumeSnapshot, bool, error) {
 	ruleName := groupSnap.Spec.PreSnapshotRule
 	if len(ruleName) == 0 {
 		groupSnap.Status.Status = stork_api.GroupSnapshotInProgress
 		// No rule, move to snapshot stage
 		groupSnap.Status.Stage = stork_api.GroupSnapshotStageSnapshot
-		return updateCRD, nil
+		return groupSnap, updateCRD, nil
 	}
 
 	log.GroupSnapshotLog(groupSnap).Infof("Running pre-snapshot rule: %s", ruleName)
 	r, err := k8s.Instance().GetRule(ruleName, groupSnap.Namespace)
 	if err != nil {
-		return !updateCRD, err
+		return groupSnap, !updateCRD, err
```

Comment:
This isn't idiomatic golang. If err is set the other return values shouldn't be used by the caller.

Reply:
fixed